pageserver/src/restore_local_repo.rs

```diff
@@ -217,10 +333,12 @@ pub fn import_timeline_wal(walpath: &Path, timeline: &dyn Timeline, startpoint:
             if rec.is_err() {
                 // Assume that an error means we've reached the end of
                 // a partial WAL record. So that's ok.
+                trace!("WAL decoder error {:?}", rec);
+                waldecoder.set_position(Lsn((segno + 1) * pg_constants::WAL_SEGMENT_SIZE as u64));
```

Comment:
What is this for?

Reply:
Switch to next segment if there are no more records in the current segment.
It is once again pg_resetwal "tricks" (also we are not using pg_resetwal utility now).\
May be we should add SWITCH record to the previous segment. But ... it means that we at least need to have this segment at pageserver. Also there may be no space for adding switch record. Certainly it can be handled. But... not done yet.